### PR TITLE
Add checking L2C capabilities mask for L2C handover

### PR DESCRIPTION
**Pull request by rge-exafore**
Requires https://github.com/swift-nav/peregrine/pull/25 to be merged first.

### CODE REVIEW

File: peregrine/gps_constants.py [hunk 1]
```diff
@@ -24,3 +24,5 @@
 
 L1CA = 'l1ca'
 L2C = 'l2c'
+
+L2C_CAPB = 0xf7814bfd
```
Comment (adel-mamin) [inline peregrine/gps_constants.py]: Let's a comment describing the constant and the reference to a place where the latest status could be looked-up. For example, https://www.unavco.org/projects/project-support/gnss-support/gnss-modernization/gnss-modernization.html

File: peregrine/run.py [hunk 2]
```diff
@@ -302,7 +307,8 @@ def main():
                                stage2_loop_filter_params=stage2_params,
                                tracker_options=tracker_options,
                                output_file=args.file,
-                               progress_bar_output=args.progress_bar)
+                               progress_bar_output=args.progress_bar,
+                               check_l2c_mask=args.check_l2c_mask)
```
Comment (adel-mamin) [inline peregrine/run.py]: Let's also add the same parameter for Tracker in peregrine/peregrine/analysis/tracking_loop.py. It is a dedicated script for tracking performance analysis.

File: peregrine/tracking.py [hunk 3]
```diff
@@ -936,14 +938,20 @@ def _create_channel(self, acq):
     """
     if not acq:
       return
+
+    l2c_handover = self.l2c_handover
+
+    if self.check_l2c_mask and (gps_constants.L2C_CAPB & (0x1 << acq.prn) == 0):
+        l2c_handover = 0
```
Comment (adel-mamin) [inline peregrine/tracking.py]: Should be 2 spaces instead of 4

**Comment by pmiettinen** [comment]
Superseded by https://github.com/swift-nav/peregrine/pull/44

**Pull request closed without merging**